hoomd/hpmc/IntegratorHPMCMonoGPU.cuh

```diff
@@ -978,7 +986,8 @@ __global__ void hpmc_insert_depletants(const Scalar4 *d_trial_postype,
         while (s_depletant_queue_size < max_depletant_queue_size && i_dep < n_depletants)
             {
             // one RNG per depletant
-            hoomd::RandomGenerator rng(hoomd::RNGIdentifier::HPMCDepletants, seed+i, i_dep, select*num_types + depletant_type, timestep);
+            hoomd::RandomGenerator rng(hoomd::Seed(hoomd::RNGIdentifier::HPMCDepletants, timestep, seed + (uint16_t)i_dep),
```

Comment:
@jglaser do we expect `n_depletants`  to be larger than 65535 at rational simulation parameters?

Reply:
that would be quite a slow simulation, but if there is a large number in the process, it would be this one. Can shuffle order of arguments once #894 gets merged

Reply:
I would prefer to have that RNG argument to have a least 32 bits

Reply:
Thanks for looking at this. It looks like these RNG are only used to generate a handful of numbers in the stream so we can safely use 16 more bits in the counter leaving only 16 bits for the stream.
How about I put the full 32-bit i_dep in the counter and then cast `select*num_types + depletant_type` down to 16 bits? Someone would either need a massive n_select or run 1 type per particle to overflow that one.

Or I could keep mixing 16-bits with the user seed but I'm not sure I like that now that I think about it. There would be some correlation between simulations with consecutive seeds. I suppose we could mix with the timestep instead - but that would lead to repeated depletion insertions after (2**24) ~16 million steps. 

`rank` also doesn't get that high in practice. How about I make `select` cast down to 16 bits and then mix the type with `rank`.